typescript/sdk/src/ism/EvmIsmModule.ts

```diff
@@ -86,4 +186,610 @@ export class EvmIsmModule extends HyperlaneModule<
+      const tx = await routingIsm.transferOwnership(delta.owner, overrides);
+      await this.multiProvider.handleTx(destination, tx);
+    }
+
+    return routingIsm;
+  }
+
+  protected async initializeFallbackRoutingIsm(params: {
+    destination: ChainName;
+    owner: Address;
+    safeConfigDomains: Domain[];
+    submoduleAddresses: Address[];
+    existingIsmAddress: Address;
+    signer: Signer;
+    logger: Logger;
+  }): Promise<IRoutingIsm> {
+    const {
+      destination,
+      owner,
+      safeConfigDomains,
+      submoduleAddresses,
+      existingIsmAddress,
+      signer,
+      logger,
+    } = params;
+    const routingIsm = DefaultFallbackRoutingIsm__factory.connect(
+      existingIsmAddress,
+      signer,
+    );
+
+    logger.debug('Initializing fallback routing ISM ...');
+    await this.multiProvider.handleTx(
+      destination,
+      routingIsm['initialize(address,uint32[],address[])'](
+        owner,
+        safeConfigDomains,
+        submoduleAddresses,
+        this.multiProvider.getTransactionOverrides(destination),
+      ),
+    );
+
+    return routingIsm;
+  }
+
+  protected async deployRoutingIsmSubmodules(params: {
+    destination: ChainName;
+    config: RoutingIsmConfig;
+    mailbox?: Address;
+  }): Promise<Address[]> {
+    const { destination, config, mailbox } = params;
+    const isms: ChainMap<Address> = {};
+
+    for (const origin of Object.keys(config.domains)) {
+      const ism = await EvmIsmModule.deploy({
+        destination,
+        config: config.domains[origin],
+        origin,
+        mailbox,
+        multiProvider: this.multiProvider,
+        factories: this.args.addresses,
+        deployer: this.deployer,
+      });
+      isms[origin] = ism.address;
+    }
+
+    return Object.values(isms);
+  }
+
+  protected static async deploy<C extends IsmConfig>(params: {
+    destination: ChainName;
+    config: C;
+    origin?: ChainName;
+    mailbox?: Address;
+    multiProvider: MultiProvider;
+    factories: HyperlaneAddresses<ProxyFactoryFactories>;
+    deployer: HyperlaneDeployer<any, any>;
+  }): Promise<DeployedIsm> {
+    const {
+      destination,
+      config,
+      origin,
+      mailbox,
+      multiProvider,
+      factories,
+      deployer,
+    } = params;
+    if (typeof config === 'string') {
+      // @ts-ignore
+      return IInterchainSecurityModule__factory.connect(
+        config,
+        multiProvider.getSignerOrProvider(destination),
+      );
+    }
+
+    const ismType = config.type;
+    const logger = rootLogger.child({ destination, ismType });
+
+    logger.debug(
+      `Deploying ${ismType} to ${destination} ${
+        origin ? `(for verifying ${origin})` : ''
+      }`,
+    );
+
+    let contract: DeployedIsmType[typeof ismType];
+    switch (ismType) {
+      case IsmType.MESSAGE_ID_MULTISIG:
+      case IsmType.MERKLE_ROOT_MULTISIG:
+        contract = await EvmIsmModule.deployMultisigIsm({
```

Comment:
any reason to assign to contract as opposed to just returning here?

Reply:
originally because there was some more work after the switch but now it's all separate fns can just return